apis/refs/v1beta1/computerefs.go

```diff
@@ -120,6 +120,102 @@ type ComputeAddressRef struct {
 	Namespace string `json:"namespace,omitempty"`
 }
 
+type ComputeAddress struct {
+	Project          string
+	Location         string
+	ComputeAddressID string
+	Address          string
+}
+
+func (c *ComputeAddress) Url() string {
+	if c.Location == "global" {
+		return fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/global/addresses/%s", c.Project, c.ComputeAddressID)
+	}
+	return fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/regions/%s/addresses/%s", c.Project, c.Location, c.ComputeAddressID)
+}
+
+func (c *ComputeAddress) GetAddress() string {
+	return fmt.Sprintf("%s", c.Address)
+}
+
+func ResolveComputeAddress(ctx context.Context, reader client.Reader, src client.Object, ref *ComputeAddressRef) (*ComputeAddress, error) {
+	if ref == nil {
+		return nil, nil
+	}
+
+	if ref.External != "" {
+		if ref.Name != "" {
+			return nil, fmt.Errorf("cannot specify both name and external on ComputeAddress reference")
+		}
+
+		tokens := strings.Split(ref.External, "/")
+		if len(tokens) == 5 && tokens[0] == "projects" && tokens[2] == "global" && tokens[3] == "addresses" {
+			return &ComputeAddress{
+				Project:          tokens[1],
+				Location:         "global",
+				ComputeAddressID: tokens[4]}, nil
+		} else if len(tokens) == 6 && tokens[0] == "projects" && tokens[2] == "location" && tokens[4] == "addresses" {
+			return &ComputeAddress{
+				Project:          tokens[1],
+				Location:         tokens[3],
+				ComputeAddressID: tokens[5]}, nil
+		}
+		return nil, fmt.Errorf("format of ComputeAddress external=%q was not known (use projects/<projectId>/global/addresses/<proxyId> or projects/<projectId>/regions/<region>/addresses/<addressId>)", ref.External)
+	}
+
+	if ref.Name == "" {
+		return nil, fmt.Errorf("must specify either name or external on ComputeAddress reference")
+	}
+
+	key := types.NamespacedName{
+		Namespace: ref.Namespace,
+		Name:      ref.Name,
+	}
+	if key.Namespace == "" {
+		key.Namespace = src.GetNamespace()
+	}
+
+	computeAddress := &unstructured.Unstructured{}
+	computeAddress.SetGroupVersionKind(schema.GroupVersionKind{
```

Comment:
Nit: would be nice to use the API GVK

Reply:
We don't have ComputeAddressGVK yet, currently no Sci-fi migration plan for this resource. I'll leave it for now and update later.